src/hooks/useClientSideV3Trade.ts

```diff
@@ -59,31 +69,49 @@ export function useClientV3TradeExactIn(
       }
     }
 
-    const { bestRoute, amountOut } = quotesResults.reduce(
-      (currentBest: { bestRoute: Route<Currency, Currency> | null; amountOut: BigNumber | null }, { result }, i) => {
+    const { bestRoute, amountIn, amountOut } = quotesResults.reduce(
+      (
+        currentBest: {
+          bestRoute: Route<Currency, Currency> | null
+          amountIn: CurrencyAmount<Currency> | null
+          amountOut: CurrencyAmount<Currency> | null
+        },
+        { result },
+        i
+      ) => {
         if (!result) return currentBest
 
-        if (currentBest.amountOut === null) {
-          return {
-            bestRoute: routes[i],
-            amountOut: result.amountOut,
+        // overwrite the current best if it's not defined or if this route is better
+        if (tradeType === TradeType.EXACT_INPUT) {
+          const amountOut = CurrencyAmount.fromRawAmount(currencyOut, result.amountOut.toString())
+          if (currentBest.amountOut === null || JSBI.lessThan(currentBest.amountOut.quotient, amountOut.quotient)) {
```

Comment:
@NoahZinsmeister curious why did you change from `CurrencyAmount.lt` to `CurrencyAmount.fromRawAmount` & `JSBI.lessThan`?

Reply:
I actually think we can just do `currentBest.amountOut.lessThan(amountOut.quotient)`

Not sure why we do this - this hook was written way back even before we used SDK frequently